src/utils/subgraph.ts

```diff
@@ -298,3 +315,29 @@ export async function getAssetsBestPrices(
 
   return assetsWithPrice
 }
+
+export async function getHighestLiquidityAssets(
+  metadataCacheUri: string,
+  sourceToken: CancelToken
+): Promise<DDO[]> {
+  const ddoList: DDO[] = []
+  const fetchedPools = await fetchData(HighestLiquidityAssets, null)
+
+  if (fetchedPools.data?.pools?.length === 0) return null
+
+  for (let i = 0; i < fetchedPools.data.pools.length; i++) {
+    const did = web3.utils
+      .toChecksumAddress(fetchedPools.data.pools[i].datatokenAddress)
+      .replace('0x', 'did:op:')
+    const ddo = await retrieveDDO(did, metadataCacheUri, sourceToken)
```

Comment:
only catching this now cause was wondering about the noticeable delay in loading. We should make this more effective by sending a list of DIDs to Aquarius in one query instead of 20 requests as it's doing right now. So the loop then is only there to construct the DID array for the Aquarius search query.